bin/hdfcoinc/pycbc_foreground_censor

```diff
@@ -28,26 +28,61 @@ pycbc.init_logging(args.verbose)
 logging.info('start')
 
 f = h5py.File(args.foreground_triggers, 'r')
-start = f['segments/foreground_veto/start'][:]
-end = f['segments/foreground_veto/end'][:]
-vsegs = pycbc.events.start_end_to_segments(start, end)
-
-ifo1, ifo2 = f.attrs['detector_1'], f.attrs['detector_2']
-
-ifos, fsegs, names = [], [], []
-for ifo in [ifo1, ifo2]:
-    segs = pycbc.events.select_segments_by_definer(args.veto_file, args.segment_name, ifo)
-    fsegs += [(segs.coalesce() + vsegs.coalesce()).coalesce()]
-    names += [args.output_segment_name]
-    ifos += [ifo]
-
-if args.strict_coinc_time:
-    fsegs[0] = (fsegs[0].coalesce() + fsegs[1].coalesce()).coalesce()
-    fsegs[1] = fsegs[0]
-
-# FIXME: Don't use URL once Larne supports taking file paths directly in the
-#        workflow File class
-file_url = urlparse.urlunparse(['file', 'localhost',
+
+if 'segments/foreground_veto/start' in f:
+    start = f['segments/foreground_veto/start'][:]
+    end = f['segments/foreground_veto/end'][:]
+    vsegs = pycbc.events.start_end_to_segments(start, end)
+
+    ifo1, ifo2 = f.attrs['detector_1'], f.attrs['detector_2']
+
+    ifos, fsegs, names = [], [], []
+    for ifo in [ifo1, ifo2]:
+        segs = pycbc.events.select_segments_by_definer(args.veto_file, args.segment_name, ifo)
+        fsegs += [(segs.coalesce() + vsegs.coalesce()).coalesce()]
+        names += [args.output_segment_name]
+        ifos += [ifo]
+
+    if args.strict_coinc_time:
+        fsegs[0] = (fsegs[0].coalesce() + fsegs[1].coalesce()).coalesce()
+        fsegs[1] = fsegs[0]
+
+    # FIXME: Don't use URL once Larne supports taking file paths directly in the
+    #        workflow File class
+    file_url = urlparse.urlunparse(['file', 'localhost',
                           os.path.abspath(args.output_file), None, None, None])
-SegFile.from_multi_segment_list('UNUSED', fsegs, names, ifos, file_url=file_url)
-logging.info('done')
+    SegFile.from_multi_segment_list('UNUSED', fsegs, names, ifos, file_url=file_url)
+else:
```

Comment:
similarly, add comment explaining the other case of the if/else switch.

Reply:
Done